fml/trace_event.cc

```diff
@@ -124,6 +124,36 @@ void TraceEventEnd(TraceArg name) {
   );
 }
 
+void TraceEventAsyncComplete(TraceArg category_group,
+                             TraceArg name,
+                             TimePoint begin,
+                             TimePoint end) {
+  auto identifier = TraceNonce();
+
+  if (begin > end) {
+    auto temp = end;
```

Comment:
use `std::swap(begin, end)`

Reply:
Hmm, I thought that needed a specialization for Timepoint.